platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/platformstatus/PlatformStatusStateMachine.java

```diff
@@ -0,0 +1,176 @@
+            } else if (action instanceof ReconnectCompleteAction reconnectCompleteAction) {
+                return currentStatusLogic.processReconnectCompleteAction(reconnectCompleteAction);
+            } else if (action instanceof SelfEventReachedConsensusAction selfEventReachedConsensusAction) {
+                return currentStatusLogic.processSelfEventReachedConsensusAction(selfEventReachedConsensusAction);
+            } else if (action instanceof StartedReplayingEventsAction startedReplayingEventsAction) {
+                return currentStatusLogic.processStartedReplayingEventsAction(startedReplayingEventsAction);
+            } else if (action instanceof StateWrittenToDiskAction stateWrittenToDiskAction) {
+                return currentStatusLogic.processStateWrittenToDiskAction(stateWrittenToDiskAction);
+            } else if (action instanceof TimeElapsedAction timeElapsedAction) {
+                return currentStatusLogic.processTimeElapsedAction(timeElapsedAction);
+            } else {
+                throw new IllegalArgumentException(
+                        "Unknown action type: " + action.getClass().getName());
+            }
+        } catch (final IllegalPlatformStatusException e) {
+            logger.error(EXCEPTION.getMarker(), e.getMessage());
+            return null;
+        }
+    }
+
+    /**
+     * Process a platform status action.
+     * <p>
+     * Repeated calls of this method cause the platform state machine to be traversed
+     *
+     * @param action the action to process
+     */
+    public void processStatusAction(@NonNull final PlatformStatusAction action) {
+        Objects.requireNonNull(action);
+
+        final PlatformStatusLogic newLogic = getNewLogic(action);
+
+        if (newLogic == null || newLogic == currentStatusLogic) {
+            // if status didn't change, there isn't anything to do
+            return;
+        }
+
+        final String previousStatusName = currentStatusLogic.getStatus().name();
+        final String newStatusName = newLogic.getStatus().name();
+
+        final String statusChangeMessage = "Platform spent %s time in %s. Now in %s"
+                .formatted(Duration.between(currentStatusStartTime, time.now()), previousStatusName, newStatusName);
+
+        logger.info(
+                PLATFORM_STATUS.getMarker(),
+                () -> new PlatformStatusPayload(statusChangeMessage, previousStatusName, newStatusName).toString());
+
+        notificationEngine.dispatch(
+                PlatformStatusChangeListener.class, new PlatformStatusChangeNotification(newLogic.getStatus()));
+
+        currentStatusLogic = newLogic;
+        currentStatusStartTime = time.now();
+    }
+
+    /**
+     * Get the current platform status
+     *
+     * @return the current platform status
+     */
+    public PlatformStatus getCurrentStatus() {
+        return currentStatusLogic.getStatus();
+    }
```

Comment:
This was written without thread consideration, since it's not yet clear whether this will be integrated with the new component framework or not

My intention was to simply deal with whichever threading considerations are necessary, based on the eventual implementation strategy

Reply:
Ok, let's just be extra careful not to forget about this.